conda-recipe/run_test.sh

```diff
@@ -48,28 +48,34 @@ if [[ "$*" == *"--json-report"* ]]; then
         rm .pytest_reports/*.json
     fi
 fi
-function json_report_name {
+
+function generate_pytest_args {
+    declare -a ARGS=()
     if [[ "${with_json_report}" == "1" ]]; then
-        printf -- "--json-report-file=.pytest_reports/$1_report.json"
+        ARGS+=("--json-report-file=.pytest_reports/$1_report.json")
+    fi
+    if [ -n "${COVERAGE_RCFILE}" ]; then
+        ARGS+=(--cov=onedal --cov=sklearnex --cov-config="${COVERAGE_RCFILE}" --cov-append --cov-report=)
     fi
+    printf -- "${ARGS[@]}"
```

Comment:
```suggestion
    printf -- "${ARGS[*]}"
```

Reply:
Looks like the change isn't needed. It produced the requisite coverage files

Reply:
Does it still produce them if you add both the json report argument and the coverage argument?

Reply:
@david-cortes-intel I checked the file and you are right that it is incomplete, I have changed it to ```"${ARGS[*]}"``` along with some other necessary changes. I will not merge without your approval because of the changes to the report generation lines.

Reply:
@david-cortes-intel I have added a matching test to the run_test.sh to match run_test.bat for checking file generation. I hope that will help address the file generation in linux question.